cli/medperf/comms/entity_resources/utils.py

```diff
@@ -60,10 +61,14 @@ def verify_or_get_hash(tmp_output_path: str, expected_hash: str) -> str:
         expected_hash (str): expected hash of the asset
 
     Returns:
-        str: Calculated hash of the asset. Only returns if the hashes match
+        str: Calculated hash of the asset. Returns None if hashes mismatch
     """
     calculated_hash = get_file_hash(tmp_output_path)
-    verify_hash(calculated_hash, expected_hash)
+    if expected_hash and calculated_hash != expected_hash:
+        logging.debug(
+            f"{tmp_output_path}: Expected {expected_hash}, found {calculated_hash}."
+        )
+        return
     return calculated_hash
```

Comment:
1. right now the function returns not str, but Optional[str]
But before fixing, look plz to the next comment at `download_resource`. This function is too complex now as it has different behavior depending on if `expected_hash` was passed; moreover, the function can be removed totally

Reply:
Agree with this sentiment